scalding-core/src/main/scala/com/twitter/package.scala

```diff
@@ -31,6 +32,8 @@ package object scalding {
   type ValuePipe[+T] = com.twitter.scalding.typed.ValuePipe[T]
   type Grouped[K, +V] = com.twitter.scalding.typed.Grouped[K, V]
 
+  implicit def typedDescriptor[T]: TypeDescriptor[T] = macro com.twitter.scalding.macros.impl.TypeDescriptorProviderImpl.caseClassTypeDescriptorImpl[T]
```

Comment:
I'm a bit nervous about this. a lot of people do `import com.twitter.scalding._`. If they also import an implicit for TypedDescriptor (as our internal code has been doing) they will now have a compile error (scala won't find either implicit, I'm pretty sure).

Maybe we want that, but I'm a bit skeptical that it is what we want. I think an explicit import might not be a bad idea, personally.